Source/Diagnostics/ReducedDiags/ParticleExtrema.cpp

```diff
@@ -0,0 +1,419 @@
+        ParallelDescriptor::ReduceRealMax(uymax);
+
+        // uzmin
+        Real uzmin = ReduceMin( myspc,
+        [=] AMREX_GPU_HOST_DEVICE (const PType& p) -> Real
+        { return p.rdata(PIdx::uz); });
+        ParallelDescriptor::ReduceRealMin(uzmin);
+
+        // uzmax
+        Real uzmax = ReduceMax( myspc,
+        [=] AMREX_GPU_HOST_DEVICE (const PType& p) -> Real
+        { return p.rdata(PIdx::uz); });
+        ParallelDescriptor::ReduceRealMax(uzmax);
+
+        // gmin
+        Real gmin = ReduceMin( myspc,
+        [=] AMREX_GPU_HOST_DEVICE (const PType& p) -> Real
+        {
+            Real ux = p.rdata(PIdx::ux);
+            Real uy = p.rdata(PIdx::uy);
+            Real uz = p.rdata(PIdx::uz);
+            Real us = ux*ux + uy*uy + uz*uz;
+            return std::sqrt(1.0 + us*inv_c2);
+        });
+        ParallelDescriptor::ReduceRealMin(gmin);
+
+        // gmax
+        Real gmax = ReduceMax( myspc,
+        [=] AMREX_GPU_HOST_DEVICE (const PType& p) -> Real
+        {
+            Real ux = p.rdata(PIdx::ux);
+            Real uy = p.rdata(PIdx::uy);
+            Real uz = p.rdata(PIdx::uz);
+            Real us = ux*ux + uy*uy + uz*uz;
+            return std::sqrt(1.0 + us*inv_c2);
+        });
+        ParallelDescriptor::ReduceRealMax(gmax);
+
+        // wmin
+        Real wmin = ReduceMin( myspc,
+        [=] AMREX_GPU_HOST_DEVICE (const PType& p) -> Real
+        { return p.rdata(PIdx::w); });
+        ParallelDescriptor::ReduceRealMin(wmin);
+
+        // wmax
+        Real wmax = ReduceMax( myspc,
+        [=] AMREX_GPU_HOST_DEVICE (const PType& p) -> Real
+        { return p.rdata(PIdx::w); });
+        ParallelDescriptor::ReduceRealMax(wmax);
+
+#if (defined WARPX_QED)
+        // compute chimin and chimax
+        Real chimin_f = 0.0_rt;
+        Real chimax_f = 0.0_rt;
+        GetExternalEField get_externalE;
+        GetExternalBField get_externalB;
+
+        if (myspc.has_breit_wheeler() || myspc.has_quantum_sync())
+        {
+
+            // declare chi arrays
+            std::vector<amrex::Real> chimin, chimax;
+            chimin.resize(level_number+1,0.0_rt);
+            chimax.resize(level_number+1,0.0_rt);
+
+            // loop over refinement levels
+            for (int lev = 0; lev <= level_number; ++lev)
+            {
+                // Loop over boxes
+                for (WarpXParIter pti(myspc, lev); pti.isValid(); ++pti)
+                {
+                    // declare position arrays
+                    Gpu::ManagedDeviceVector<ParticleReal> x, y, z;
+                    // get particle arrays
+                    pti.GetPosition(x, y, z);
+                    auto & w  = pti.GetAttribs(PIdx::w);
+                    auto & ux = pti.GetAttribs(PIdx::ux);
+                    auto & uy = pti.GetAttribs(PIdx::uy);
+                    auto & uz = pti.GetAttribs(PIdx::uz);
+                    // declare external fields
+                    int offset = 0;
```

Comment:
```suggestion
                    const int offset = 0;
```
Could this be `const` ?